crate_universe/src/context/platforms.rs

```diff
@@ -165,39 +165,58 @@ mod test {
         assert_eq!(configurations, BTreeMap::new(),)
     }
 
-    #[test]
-    fn resolve_targeted() {
-        let configuration = r#"cfg(target = "x86_64-unknown-linux-gnu")"#.to_owned();
+    fn mock_resolve_context(configuration: String) -> CrateContext {
         let mut deps = SelectList::default();
         deps.insert(
             CrateDependency {
                 id: CrateId::new("mock_crate_b".to_owned(), "0.1.0".to_owned()),
                 target: "mock_crate_b".to_owned(),
                 alias: None,
             },
-            Some(configuration.clone()),
+            Some(configuration),
         );
 
-        let context = CrateContext {
+        CrateContext {
             name: "mock_crate_a".to_owned(),
             version: "0.1.0".to_owned(),
             common_attrs: CommonAttributes {
                 deps,
                 ..CommonAttributes::default()
             },
             ..CrateContext::default()
-        };
+        }
+    }
 
-        let configurations =
-            resolve_cfg_platforms(vec![&context], &supported_platform_triples()).unwrap();
+    #[test]
+    fn resolve_targeted() {
```

Comment:
This test was refactored to gain higher confidence in it's validity. The real issue I was seeing was addressed by https://github.com/bazelbuild/rules_rust/pull/1591#discussion_r992959641